sdk/typespec/typespec_client_core/src/http/sanitizer.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore traceparent tracestate
+
+use crate::http::{headers::Headers, Url};
+use std::{collections::HashSet, sync::LazyLock};
+
+pub static DEFAULT_ALLOWED_HEADER_NAMES: LazyLock<HashSet<&'static str>> = LazyLock::new(|| {
```

Comment:
Comment. But also, if you are using the string values directly, why did you make the headers `pub(crate)`? Again, though, I'd just remove that change and let's remove all but the helpful ones that help parse types as part of a separate work item. Keep this one focused.